tooling/bundler/src/bundle/linux/rpm.rs

```diff
@@ -5,10 +5,147 @@
 
 use crate::Settings;
 
-use std::path::PathBuf;
+use anyhow::Context;
+use log::info;
+use rpm::{self, signature::pgp, Dependency, FileMode, FileOptions};
+use std::{
+  env,
+  fs::{self, File},
+  path::{Path, PathBuf},
+};
+
+use super::freedesktop;
 
 /// Bundles the project.
-/// Not implemented yet.
-pub fn bundle_project(_settings: &Settings) -> crate::Result<Vec<PathBuf>> {
-  unimplemented!();
+/// Returns a vector of PathBuf that shows where the RPM was created.
+pub fn bundle_project(settings: &Settings) -> crate::Result<Vec<PathBuf>> {
+  let name = settings.main_binary_name();
+  let version = settings.version_string();
+  let release = settings.rpm().release.as_str();
+  let epoch = settings.rpm().epoch;
+  let arch = match settings.binary_arch() {
+    "x86" => "i386",
+    "arm" => "armhfp",
+    other => other,
+  };
+
+  let license = settings
+    .rpm()
+    .license
+    .as_deref()
+    .or_else(|| settings.license())
+    .unwrap_or_default();
+
+  let summary = settings.short_description().trim();
+
+  let package_base_name = format!("{name}-{version}-{release}.{arch}");
+  let package_name = format!("{package_base_name}.rpm");
+
+  let base_dir = settings.project_out_directory().join("bundle/rpm");
+  let package_dir = base_dir.join(&package_base_name);
+  if package_dir.exists() {
+    fs::remove_dir_all(&package_dir)
+      .with_context(|| format!("Failed to remove old {package_base_name}"))?;
+  }
+  fs::create_dir_all(&package_dir)?;
+  let package_path = base_dir.join(&package_name);
+
+  info!(action = "Bundling"; "{} ({})", package_name, package_path.display());
+
+  let mut builder = rpm::PackageBuilder::new(name, version, license, arch, summary)
+    .epoch(epoch)
+    .release(release);
```

Comment:
Can you add this in this PR too? https://github.com/tauri-apps/tauri/pull/9840/files